### ISSUE TITLE
Auto refresh market info after new orders coming.

### ISSUE THREAD

**Issue by McGrady00H** [peatio/peatio#569]
Hi all~  Finally I can run the exchange on my Mac.  Now I have a question about the market info ( e.g. http://127.0.0.1:3000/markets/btccny) . The market info is expected to refresh automatically when a new order comes or some trading events happens. ( e.g. when I submit an order, the it will appear in the following place automatically, however, I need to refresh the website by hand)
![image](https://user-images.githubusercontent.com/5101761/27725854-c2d0851e-5daa-11e7-9477-f5df3a8535d6.png). Is this the default setting or feature missing for Peatio
Thanks!

**Comment by VanillaPete** [comment]
Likely an issue with your pusher/slanger setup.

**Comment by VanillaPete** [comment]
If anyone needs help setting up Peatio I will have some free time in the coming months. Email me at exmakerpete@gmail.com.

**Comment by amarian12** [comment]
i do have same issue using slanger, everything works perfect excepting the fact i need to refresh manually the browser to updates the #order_book and #my_orders fields when i place a new order (buy or sell)

**Comment by khubab** [comment]
I have peatio configuration and setup experience. You can contact me at khubabmazhar596@hotmail.com and I will help you with auto-order update issue.

**Comment by amarian12** [comment]
i fix it myself ... the firewall was the problem

**Comment by khubab** [comment]
It can't be pusher issue as pusher also connect and get stats on refreshing page. As you have mentioned that it was firewall issue. Can you specify which port has issue?

**Comment by amarian12** [comment]
i use slanger so i can select the port to use (default it's 4567)  ... but i forgot to make the firewall rule to allow that port :(

./slanger --app_key key --secret secret --redis_address redis://127.0.0.1:6379 -a 0.0.0.0:4567 -w 0.0.0.0:8080

all ports are configurable

**Comment by khubab** [comment]
slanger is for development version only and one shouldn't prefer it in production environment. The command you have mentioned is to start slanger instance. It has nothing to do with the issue. In production environment, I have experienced that prices are not updating automatically. so, if you are going to play then slanger is fine but it's not for running an exchange. Firewall rule will not block slanger on local system as requests are originating and terminating on the same system.

**Comment by khubab** [comment]
slanger is for development version only and one shouldn't prefer it in production environment. The command you have mentioned is to start slanger instance. It has nothing to do with the issue. In production environment, I have experienced that prices are not updating automatically. so, if you are going to play then slanger is fine but it's not for running an exchange. Firewall rule will not block slanger on local system as requests are originating and terminating on the same system.

**Comment by amarian12** [comment]
you are correct, but i do not run on on same machine, i do have more servers running all services separate from each other. 

I do prefer slanger as I can manage myself the full environment i do not depend on external sources.

**Comment by rgcosta** [comment]
@McGrady00H Did you solve your problem? How?

@amarian12 could you guide me how to set slanger on peatio?

**Comment by scatterp** [comment]
seems like you need to check redis server rgcosta verify its installed and working... maybe update it to latest version verify your pusher settings are in application.yml and also that your pusher free limit (check your mail) has not been passed tell me the result (in chat)

**Comment by legends420** [comment]
any fix? cant get it to auto refresh

**Comment by scatterp** [comment]
legends did you read anything here in this thread... yes the answer is here .. your pusher settings

**Comment by legends420** [comment]
yes i did the pusher settings then precompiled again and still nothing

**Comment by amarian12** [comment]
to fix pusher check this patch :

https://github.com/rubykube/peatio/commit/1a23ec09b98bf66f8b371bdd3374e7f335f19172

**Comment by legends420** [comment]
Fix : Make sure when you create PUSHER APP you pick MTN1 CLUSTER as that is what is default and you will not have any issues

**Comment by scatterp** [comment]
correct it think its us east

**Comment by Matchek79** [comment]
Hi, got the same problem when using slanger market do not autorefresh, any advice ?

**Comment by Matchek79** [comment]
It' seems that, sleanger is deployed (localhost) but some part of connection is goin to pusher, I still see at pusher connections from my host. application.yml is pointing to localhost. So why pusher shows connections ?

**Comment by amarian12** [comment]
in peatio application.yml you don;t use localhost ... you need to use your dns or ip and also you need to open the ports for slanger

**Comment by amarian12** [comment]
yes you can run it on same machine, but you need to have ports mapped so external clients can connect to it:

slanger --app_key appkey --secret secretkey --redis_address redis://127.0.0.1:6379 -a YourDNS:PORT1 -w YourDNS:PORT2

PORT1 and PORT2 need to be nated.


you can use localhost, but will work only for localhost connection (when you browse locally the site):

slanger --app_key appkey --secret secretkey --redis_address redis://127.0.0.1:6379 -a localhost:PORT1 -w localhost:PORT2

also this 2 ports need to be defined in peatio also.

**Comment by Matchek79** [comment]
Still nothing. At which files I have to change pusher host name?
done at config/application.yml 

![image](https://user-images.githubusercontent.com/6608603/38296629-2e79b12e-37f2-11e8-9805-ff79767f387f.png)

**Comment by amarian12** [comment]
Local setup only: 

 PUSHER_APP: 893273
  PUSHER_KEY: appkey
  PUSHER_SECRET: secretkey
  PUSHER_HOST: localhost
  PUSHER_PORT: PORT1
  PUSHER_WS_PORT: PORT2
  PUSHER_WSS_PORT: PORT2
  PUSHER_ENCRYPTED: false
  
  
slanger --app_key appkey --secret secretkey --redis_address redis://127.0.0.1:6379 -a localhost:PORT1 -w localhost:PORT2

**Comment by Matchek79** [comment]
@amarian12 check info@e-pool.net I've sent you msg
